vendor/blockchain_contracts/src/bin/calculate_offsets/ethereum/rfc003/contract.rs

```diff
@@ -0,0 +1,79 @@
+extern crate regex;
+use crate::calculate_offsets::ethereum::rfc003::{
+    compile_contract::compile, metadata::Metadata, offset::Offset,
+    placeholder_config::PlaceholderConfig, Error,
+};
+
+const TEMPLATE_FOLDER: &str = "./src/bin/calculate_offsets/ethereum/rfc003/templates/";
+const CONTRACT_FILE: &str = "contract.asm";
+const HEADER_FILE: &str = "deploy_header.asm";
+const CONFIG_FILE: &str = "config.json";
+
+pub struct Contract {
+    bytes: Vec<u8>,
+    placeholder_config: PlaceholderConfig,
+    pub meta_data: Metadata,
+}
+
+impl Contract {
+    pub fn compile_from_directory_and_load_placeholder_config(
+        dir: &'static str,
+    ) -> Result<Contract, Error> {
+        let mut bytes = compile(&get_header_file_path())?;
+        let mut contract_body = compile(&get_contract_file_path(dir))?;
+        bytes.append(&mut contract_body);
+
+        let placeholder_config = PlaceholderConfig::from_file(&get_config_file_path(dir))?;
+
+        let meta_data = Metadata::new(
+            placeholder_config.ledger_name.to_owned(),
+            placeholder_config.ledger_name.to_owned(),
+            hex::encode(bytes.to_owned()),
```

Comment:
:wrench: This could be done lazily as you need it. Also, there should be no need for `to_owned` since it is already a `Vec<u8>`.

Reply:
The `to_owned` is somehow still needed, let's take a look at this together

Reply:
conclusion: `Metadata` only contains Strings as it represents the metadata to be printed. So it's fine to stay like this :)